build-tools/automation/azure-pipelines.yaml

```diff
@@ -781,6 +781,11 @@ stages:
         git submodule update -q --init --recursive
       displayName: Clone and update UITools
 
+    - task: NuGetAuthenticate@0
```

Comment:
What are the conditions to use the **NuGetAuthenticate** task? In other words, when should we use that task and when it is not needed?

Reply:
This was needed to access the package source used in android-sdk-installer: https://github.com/xamarin/android-sdk-installer/commit/e2f249ea010ad266861acb21890b65374cfc6797#diff-b54ee602750f150b3d4986e0334cc3ec901d47510560e26e792e8d294ab6f37aR5

@joj can maybe help provide more information about why it's now needed, I am assuming this new source is more restricted some how.